addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -1,163 +1,145 @@
 odoo.define('mail.ActivityRenderer', function (require) {
 "use strict";
 
-var AbstractRenderer = require('web.AbstractRenderer');
-var ActivityRecord = require('mail.ActivityRecord');
-var config = require('web.config');
-var core = require('web.core');
-var field_registry = require('web.field_registry');
-var KanbanColumnProgressBar = require('web.KanbanColumnProgressBar');
-var qweb = require('web.QWeb');
-var session = require('web.session');
-var utils = require('web.utils');
-
-var KanbanActivity = field_registry.get('kanban_activity');
-var _t = core._t;
-var QWeb = core.qweb;
-
-var ActivityRenderer = AbstractRenderer.extend({
-    className: 'o_activity_view',
-    custom_events: {
-        'set_progress_bar_state': '_onSetProgressBarState',
-    },
-    events: {
-        'click .o_send_mail_template': '_onSenMailTemplateClicked',
-        'click .o_activity_empty_cell': '_onEmptyCell',
-        'click .o_record_selector': '_onRecordSelector',
-    },
-
-    /**
-     * @override
-     * @param {Object} params.templates
-     */
-    init: function (parent, state, params) {
-        this._super.apply(this, arguments);
-
-        this.qweb = new qweb(config.isDebug(), {_s: session.origin});
-        this.qweb.add_template(utils.json_node_to_xml(params.templates));
-    },
+const AbstractRendererOwl = require('web.AbstractRendererOwl');
+const ActivityCell = require('mail.ActivityCell');
+const ActivityRecord = require('mail.ActivityRecord');
+const { ComponentAdapter } = require('web.OwlCompatibility');
+const config = require('web.config');
+const core = require('web.core');
+const KanbanColumnProgressBar = require('web.KanbanColumnProgressBar');
+const QWeb = require('web.QWeb');
+const session = require('web.session');
+const utils = require('web.utils');
+
+const _t = core._t;
+
+const { useState } = owl.hooks;
+
+/**
+ * Owl Component Adapter for ActivityRecord which is KanbanRecord (Odoo Widget)
+ * TODO: Remove this adapter when ActivityRecord is a Component
+ */
+class ActivityRecordAdapter extends ComponentAdapter {
+    render() {
+        this.widget._render();
+    }
+
+    update(nextProps) {
+        const state = nextProps.widgetArgs[0];
+
+        // This code comes from KanbanRecord.update().
+        // Here we do not re-render.
+        _.invoke(_.pluck(this.widget.subWidgets, '$el'), 'detach');
```

Comment:
you kinda re-render (partially at least, as you detach subwidgets from the DOM). This should be done just before the call to _render I think